### PR TITLE
Handle readers in folder name

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

This adds a new field to our `UserList` type to keep track of readers

### CODE REVIEW

File: shared/actions/favorite.js [hunk 1]
```diff
@@ -9,12 +9,14 @@ import type {UserList} from '../common-adapters/usernames'
...
 import {badgeApp} from './notifications'
-import {canonicalizeUsernames, parseFolderNameToUsers} from '../util/kbfs'
+import {parseFolderNameToUsers, sortUserList} from '../util/kbfs'
 import {defaultKBFSPath} from '../constants/config'
 import {navigateBack} from '../actions/router'
 
 export function pathFromFolder ({isPublic, users}: {isPublic: boolean, users: UserList}) {
-  const sortName = users.map(u => u.username).join(',')
+  const rwers = users.filter(u => !u.readOnly).map(u => u.username)
+  const readers = users.filter(u => !!u.readOnly).map(u => u.username)
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: I know I can use `_.partition` but didn't want to import it, if there's demand I'll change it

File: shared/util/kbfs.js [hunk 2]
```diff
@@ -1,14 +1,34 @@
 /* @flow */
+import type {UserList} from '../common-adapters/usernames'
 
-// Parses the folder name and returns an array of usernames (TODO: handle read only-ers)
-export function parseFolderNameToUsers (folderName: string): Array<string> {
-  return folderName.split(',')
+// Parses the folder name and returns an array of usernames
+export function parseFolderNameToUsers (yourUsername: ?string, folderName: string): UserList {
+  const [rwers, readers = ''] = folderName.split('#')
```
Comment (chromakode) [inline shared/util/kbfs.js]: Wow such destructuring

File: shared/util/kbfs.js [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
+  return rwers.split(',')
```
Comment (chromakode) [inline shared/util/kbfs.js]: :sheep: Might be a little more readable to break this down into `const` assignments for the individual array steps

**Comment by chromakode** [comment]
:+1:

**Comment by chromakode** [comment]
:+1:

**Pull request merged**